### PR TITLE
Adjust RTC environmental variables.

### PR DESCRIPTION
**Pull request by ptheywood**
+ Checks for `CUDA_PATH` and then `CUDA_HOME` to find the cuda include directory.
+ Checks for `FLAMEGPU_INC_DIR` and then `FLAMEGPU2_INC_DIR` for the flamegpu include direcotry

Part of #253

### CODE REVIEW

File: src/flamegpu/util/detail/JitifyCache.cu [hunk 1]
```diff
@@ -81,74 +82,138 @@ std::string loadFile(const path &filepath) {
     ifs.close();
     return rtn;
 }
-}  // namespace
 
-std::mutex JitifyCache::instance_mutex;
```
Comment (Robadob) [inline src/flamegpu/util/detail/JitifyCache.cu:86]: Doc comments go in the header pls

File: src/flamegpu/util/detail/JitifyCache.cu [hunk 2]
```diff
@@ -81,74 +82,138 @@ std::string loadFile(const path &filepath) {
...
-    static std::string env_cuda_path = std::getenv("CUDA_PATH") ? std::getenv("CUDA_PATH") : "";
-    if (env_inc_fgp2.empty()) {
+/**
+ * Find the cuda include directory.
+ * Throws exceptions if it can not be found.
+ * @return the path to the CUDA include directory.
+ */
+std::string getCUDAIncludeDir() {
+    // Define an array of environment variables to check in order
+    std::array<const std::string, 2> ENV_VARS { "CUDA_PATH", "CUDA_HOME" };
```
Comment (Robadob) [inline src/flamegpu/util/detail/JitifyCache.cu:93]: This feels like something vs gives warnings about (don't think it likes const std::string in collections for some reason).

**Review by Robadob — APPROVED** [review]
Looks good.

File: src/flamegpu/util/detail/JitifyCache.cu (hunk 2 above)
Comment (ptheywood) [reply src/flamegpu/util/detail/JitifyCache.cu:93]: CI didn't seem to trigger so it looks like this isnt' quite what MSVC gets angry about, but I do seem to recall something along these lines being problematic in the past.

**Pull request merged**